src/contrib/mlir/lowerer.cpp

```diff
@@ -534,91 +534,72 @@ namespace
     {
         auto concat = cast<NGConcatOp>(op);
         auto loc = concat.getLoc();
-
         ScopedContext scope(rewriter, loc);
 
         // Create Value for result, and extract type info.
         Value* result = m_pass.buildOutputDefs(op, rewriter)[0];
         NGRAPH_CHECK(result, "Unexpected null result in ConcatOp");
+        auto resultTy = result->getType().dyn_cast<MemRefType>();
+        NGRAPH_CHECK(resultTy, "Unexpected non-memref result type");
 
-        auto result_ty = result->getType().dyn_cast<MemRefType>();
-        NGRAPH_CHECK(result_ty, "Unexpected non-memref result type");
-
-        Type elem_ty = result_ty.getElementType();
-
-        // Retrieve concatenation axis.
-        auto concatenation_axis_attr = op->getAttrOfType<IntegerAttr>("concatenation_axis");
-        NGRAPH_CHECK(concatenation_axis_attr, "Missing concatenation_axis in ConcatOp");
-
-        int64_t concatenation_axis = concatenation_axis_attr.getInt();
-
-        // Create views/handles to write into result.
-        MemRefView v_res(result);
-        auto rank = v_res.rank();
-        IndexedValue iv_res(result);
+        // Create view to write into result.
+        MemRefView vRes(result);
+        auto rank = vRes.rank();
 
         // For each operand, generate a separate loop to copy into the target slice of "result".
         // We'll keep track of the slice offsets via concatenation_axis_pos.
-        IndexHandle concatenation_axis_pos(index_t(0));
+        auto concatenationAxis = concat.concatenation_axis().getSExtValue();
+        IndexHandle concatenationAxisPos(index_t(0));
 
         for (auto& operand : operands)
```

Comment:
nit: I think operand has pointer type, better use * instead of reference?

Reply:
The `auto&` here will give us references to the pointers, so I think this should be okay.